### ISSUE TITLE
Remove the ability to send to Sprout addresses from RPC calls

### ISSUE THREAD

**Issue by jackgavigan** [zcash/zcash#3773]
Remove the ability to send transactions to Sprout shielded addresses from zcashd in order to reduce usage of the Sprout shielded pool.

This could be preceded by a change to _discourage_ sending to Sprout addresses by requiring that the user "click-through" a warning message that they should encourage whoever they are sending funds to to create a Sapling address. However, such a change requires assessment from a usability perspective. 

This is the logical next step after #3752, and a precursor to ~#3771~ #3677.

**Comment by daira** [comment]
This can be merged into #4557.
